internal/api/import.go

```diff
@@ -0,0 +1,112 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/awat/model"
+)
+
+// handleStartImport handles POST requests sent to /import This
+// endpoint takes an ID, locks the oldest Import awaiting work with
+// that ID, and returns the metadata associated with that Import in
+// order for a Provisioner to being work on it
+func handleStartImport(c *Context, w http.ResponseWriter, r *http.Request) {
+	workRequest, err := model.NewImportWorkRequestFromReader(r.Body)
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to unmarshal request for work")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	work, err := c.Store.GetAndClaimNextReadyImport(workRequest.ProvisionerID)
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to fetch import")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if work == nil {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	outputJSON(c, w, importStatusFromImport(work))
+}
+
+// handleListImports responds to GET /imports and returns all Imports
+// in the database
+// TODO add pagination to this endpoint
+func handleListImports(c *Context, w http.ResponseWriter, r *http.Request) {
+	imports, err := c.Store.GetAllImports()
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to fetch imports")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	if len(imports) < 1 {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
```

Comment:
0/5, but just returning the empty imports list is good as well. That's how the provisioner behaves if you would like to keep behavior similar.